Manifest controls (#3223)
OfficeDev/office-js-docs-pr: docs/reference/manifest/control-menu.md

---

> title: Menu type Control element in the manifest file

Reviewer: ```suggestion
title: Control element of type Menu in the manifest file
```

> description: Defines a menu whose items can execute an actions or launch task panes.

Reviewer: ```suggestion
description: Defines a menu whose items can execute actions or launch task panes.
```

ms.date: 02/04/2022
ms.localizationpriority: medium
---

# Control element of type Menu

A menu defines a list of options. Each menu item either executes a function or shows a task pane.

> [!NOTE]
> This article assumes familiarity with the basic [Control reference article](control.md) which contains important information about the element's attributes.

The menu control defines:

- A root-level menu control.
- A list of menu items.

> When used with **PrimaryCommandSurface** [extension point](extensionpoint.md), the root menu item displays as a button on the ribbon. When the button is selected, the menu displays as a drop-down list. Submenus are not supported.

Reviewer: Per the [MS Style Guide](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28800), should be dropdown
```suggestion
When used with the **PrimaryCommandSurface** [extension point](extensionpoint.md), the root menu item displays as a button on the ribbon. When the button is selected, the menu displays as a dropdown list. Submenus are not supported.
```
